pkg/util/names/names.go

```diff
@@ -91,6 +91,13 @@ func GenerateWorkName(kind, name, namespace string) string {
 		name = strings.ReplaceAll(name, ":", ".")
 	}
 
+	// The name of resources, like 'Role'/'ClusterRole'/'RoleBinding'/'ClusterRoleBinding',
+	// may contain upper case characters that are not allowed by CRD resources which require the
+	// name can be used as a DNS subdomain name. So, we need to replace it.
+	// For more information about the DNS subdomain name, please refer to
+	// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names.
+	name = strings.ToLower(name)
```

Comment:
Given these exceptions are all caused by RBAC resources, we don't need to repeat the reasons, we can extend the comments like:
```diff
diff --git a/pkg/util/names/names.go b/pkg/util/names/names.go                                                          
index 5d04a48b..791ce90c 100644                                                                                         
--- a/pkg/util/names/names.go                                                                                           
+++ b/pkg/util/names/names.go                                                                                           
@@ -81,7 +81,7 @@ func GenerateBindingReferenceKey(namespace, name string) string {                                     
 // GenerateWorkName will generate work name by its name and the hash of its namespace, kind and name.                  
 func GenerateWorkName(kind, name, namespace string) string {                                                           
        // The name of resources, like 'Role'/'ClusterRole'/'RoleBinding'/'ClusterRoleBinding',                         
-       // may contain symbols(like ':') that are not allowed by CRD resources which require the                        
+       // may contain symbols(like ':' or uppercase upper case) that are not allowed by CRD resources which require the
        // name can be used as a DNS subdomain name. So, we need to replace it.                                         
        // These resources may also allow for other characters(like '&','$') that are not allowed                       
        // by CRD resources, we only handle the most common ones now for performance concerns.                          
@@ -90,6 +90,7 @@ func GenerateWorkName(kind, name, namespace string) string {
        if strings.Contains(name, ":") {
                name = strings.ReplaceAll(name, ":", ".")
        }
+       name = strings.ToLower(name)

        var workName string
        if len(namespace) == 0 {

```

Reply:
> Given these exceptions are all caused by RBAC resources, we don't need to repeat the reasons, we can extend the comments like:
> 
> ```diff
> diff --git a/pkg/util/names/names.go b/pkg/util/names/names.go                                                          
> index 5d04a48b..791ce90c 100644                                                                                         
> --- a/pkg/util/names/names.go                                                                                           
> +++ b/pkg/util/names/names.go                                                                                           
> @@ -81,7 +81,7 @@ func GenerateBindingReferenceKey(namespace, name string) string {                                     
>  // GenerateWorkName will generate work name by its name and the hash of its namespace, kind and name.                  
>  func GenerateWorkName(kind, name, namespace string) string {                                                           
>         // The name of resources, like 'Role'/'ClusterRole'/'RoleBinding'/'ClusterRoleBinding',                         
> -       // may contain symbols(like ':') that are not allowed by CRD resources which require the                        
> +       // may contain symbols(like ':' or uppercase upper case) that are not allowed by CRD resources which require the
>         // name can be used as a DNS subdomain name. So, we need to replace it.                                         
>         // These resources may also allow for other characters(like '&','$') that are not allowed                       
>         // by CRD resources, we only handle the most common ones now for performance concerns.                          
> @@ -90,6 +90,7 @@ func GenerateWorkName(kind, name, namespace string) string {
>         if strings.Contains(name, ":") {
>                 name = strings.ReplaceAll(name, ":", ".")
>         }
> +       name = strings.ToLower(name)
> 
>         var workName string
>         if len(namespace) == 0 {
> ```

I'll merge those comments, and is there any other work that needs to be modified? Such as ut?

Reply:
Just this part. I'm good with the current ut.

Reply:
> Just this part. I'm good with the current ut.

Okay, I've finished.